lib/view_component/config.rb

```diff
@@ -0,0 +1,158 @@
+
+      # @!attribute preview_controller
+      # @return [String]
+      # The controller used for previewing components.
+      # Defaults to `ViewComponentsController`.
+
+      # @!attribute preview_route
+      # @return [String]
+      # The entry route for component previews.
+      # Defaults to `"/rails/view_components"`.
+
+      # @!attribute show_previews_source
+      # @return [Boolean]
+      # Whether to display source code previews in component previews.
+      # Defaults to `false`.
+
+      # @!attribute instrumentation_enabled
+      # @return [Boolean]
+      # Whether ActiveSupport notifications are enabled.
+      # Defaults to `false`.
+
+      # @!attribute render_monkey_patch_enabled
+      # @return [Boolean] Whether the #render method should be monkey patched.
+      # If this is disabled, use `#render_component` or
+      # `#render_component_to_string` instead.
+      # Defaults to `true`.
+
+      # @!attribute view_component_path
+      # @return [String]
+      # The path in which components, their templates, and their sidecars should
+      # be stored.
+      # Defaults to `"app/components"`.
+
+      # @!attribute component_parent_class
+      # @return [String]
+      # The parent class from which generated components will inherit.
+      # Defaults to `nil`. If this is falsy, generators will use
+      # `"ApplicationComponent"` if defined, `"ViewComponent::Base"` otherwise.
+
+      # @!attribute show_previews
+      # @return [Boolean]
+      # Whether component previews are enabled.
+      # Defaults to `true` in development and test environments.
+
+      # @!attribute use_global_output_buffer
+      # @return [Boolean]
+      # Whether to use the experimental global output buffer for all components.
+      # It can be used on a per-component basis with `prepend
+      # ViewComponent::GlobalOutputBuffer.`
+      # Defaults to `false`.
+
+      # @!attribute preview_paths
+      # @return [Array<String>]
+      # The locations in which component previews will be looked up.
+      # Defaults to `['test/component/previews']` relative to your Rails root.
+
+      # @!attribute preview_path
+      # @deprecated Use #preview_paths instead. Will be removed in v3.0.0.
+
+      # @!attribute test_controller
+      # @return [String]
+      # The controller used for testing components.
+      # Can also be configured on a per-test basis using `#with_controller_class`.
+      # Defaults to `ApplicationController`.
+
+      # @!attribute default_preview_layout
+      # @return [String]
+      # A custom default layout used for the previews index page and individual
+      # previews.
+      # Defaults to `nil`. If this is falsy, `"component_preview"` is used.
+
+      def default_preview_paths
+        return [] unless defined?(Rails.root) && Dir.exist?("#{Rails.root}/test/components/previews")
+
+        ["#{Rails.root}/test/components/previews"]
+      end
+    end
+
+    def initialize(parent = nil)
```

Comment:
I think I had to keep the method signature this way for all cases. Maybe `ActiveSupport::Configurable` tries to use it? It might hinder composing around `InheritableOptions` as you mentioned elsewhere, but I'll investigate further and give it a shot.